fetch-in-react/src/app.jsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import UserList from './user-list';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      users: [],
+      isLoading: true
+    };
+  }
+
+  componentDidMount() {
+    fetch('https://jsonplaceholder.typicode.com/users')
+      .then(res => {
+        if (!res.ok) return console.error(res); else return res.json();
+      })
+      .then(res => {
+        this.setState(prev => (
+          {
+            users: res,
```

Comment:
Careful, if you `return console.error(res)` above, this `res` will be undefined. Here's what I'd recommend instead

```js
fetch('...')
  .then(res => {
    return res.json()
      .then(data => {
          if (res.ok) return { users: data, error: null }
          return { users: [], error: data }
      })
      .then(result => {
        this.setState({ ...result, isLoading: false })
      })
      .catch(err => console.error(err))
  })
```